homeassistant/components/tedee/__init__.py

```diff
@@ -0,0 +1,80 @@
+"""Init the tedee component."""
+import logging
+
+from pytedee_async import TedeeAuthException, TedeeClientException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.network import NoURLAvailableError, get_url
+
+from .const import CONF_HOME_ASSISTANT_ACCESS_TOKEN, DOMAIN
+from .coordinator import TedeeApiCoordinator
+from .views import TedeeWebhookView
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+PLATFORMS = [
+    Platform.LOCK,
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Integration setup."""
+    entry.async_on_unload(entry.add_update_listener(options_update_listener))
+
+    hass.data.setdefault(DOMAIN, {})[
+        entry.entry_id
+    ] = coordinator = TedeeApiCoordinator(hass, entry)
```

Comment:
Please assign the coordinator to hass.data after you know init was successful
```suggestion
coordinator = TedeeApiCoordinator(hass, entry)
```

```python
...
await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```

Reply:
I personally have no preference here, but it was requested like I have it from another reviewer in my other PR.

Reply:
fixed